features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/SearchFragment.kt

```diff
@@ -285,240 +305,212 @@ class SearchFragment : Fragment(R.layout.fragment_search) {
+        bottomSheet.expandedOffset = resources.getDimensionPixelOffset(R.dimen.default_expanded_offset)
 
         bottomSheet.state = if (bottomSheetWasExpanded) {
             BottomSheetBehavior.STATE_EXPANDED
         } else {
             BottomSheetBehavior.STATE_HALF_EXPANDED
         }
 
+        if (binding.searchResults.itemDecorationCount < 1) {
+            binding.searchResults.addItemDecoration(searchResultsDecorator)
+        }
+
+        binding.searchResults.adapter = ConcatAdapter(searchHeaderAdapter, searchResultsAdapter)
+        val layoutParams = binding.searchResults.layoutParams as ConstraintLayout.LayoutParams
+        layoutParams.topMargin = resources.getDimensionPixelOffset(R.dimen.search_results_margin_top)
         binding.searchResults.isVisible = true
         binding.dragIndicator.isVisible = true
 
         val animResults = ObjectAnimator.ofFloat(binding.searchResults, View.ALPHA, 1f)
         val animDrag = ObjectAnimator.ofFloat(binding.dragIndicator, View.ALPHA, 1f)
-        val animDetails = ObjectAnimator.ofFloat(binding.itemDetails.root, View.ALPHA, 0f)
+        val animDetails = ObjectAnimator.ofFloat(binding.itemDetails, View.ALPHA, 0f)
         AnimatorSet().apply {
             playTogether(animResults, animDrag, animDetails)
             duration = 200
             doOnEnd {
-                binding.itemDetails.root.isVisible = false
+                binding.itemDetails.isVisible = false
+                binding.filterPanel.isVisible = shouldShowFiltersPanel()
+            }
+        }.start()
+    }
+
+    private fun transitToAllMerchantLocations(expand: Boolean) {
+        Log.i("EXPLOREDASH", "transit to all merchant locations: ${viewModel.selectedItem}")
+        binding.upButton.isVisible = shouldShowUpButton()
+        binding.filterPanel.isVisible = false
+
+        val canShowNearest = viewModel.canShowNearestLocation()
+        binding.backToNearestBtn.isVisible = canShowNearest
+
+        val bottomSheet = BottomSheetBehavior.from(binding.contentPanel)
+        bottomSheet.isDraggable = viewModel.isLocationEnabled.value == true
+        bottomSheet.expandedOffset = resources.getDimensionPixelOffset(R.dimen.all_locations_expanded_offset)
+        bottomSheet.state = if (viewModel.isLocationEnabled.value != true || expand)
+        {
+            BottomSheetBehavior.STATE_EXPANDED
+        } else {
+            BottomSheetBehavior.STATE_HALF_EXPANDED
+        }
+
+        viewModel.selectedItem.value?.let { item ->
+            val header = MerchantLocationsHeaderAdapter(
+                item.name ?: "",
+                binding.itemDetails.getMerchantType(item.type),
+                item.logoLocation ?: ""
+            )
+
+            if (binding.searchResults.itemDecorationCount > 0) {
+                binding.searchResults.removeItemDecorationAt(0)
+            }
+
+            binding.searchResults.adapter = ConcatAdapter(header, merchantLocationsAdapter)
```

Comment:
Because of BottomSheet behaviour issues, we have to use one RecyclerView for both search results and merchant locations.